packages/embark-ui/src/containers/TransactionsContainer.js

```diff
@@ -34,34 +33,56 @@ class TransactionsContainer extends Component {
   }
 
   get numberOfBlocks() {
-    const blocks = this.props.blocks;
-    return !blocks.length ? 0 : blocks[0].number + 1;
+    if (this._numberOfBlocks === undefined) {
+      const blocks = this.props.blocks;
+      this._numberOfBlocks = !blocks.length ? 0 : blocks[0].number + 1;
+    }
+    return this._numberOfBlocks;
   }
 
   get estNumberOfTxs() {
-    const blocks = this.props.blocks;
-    const numBlocksInProps = blocks.length;
-    const numTxsInPropsBlocks = blocks.reduce((txCount, block) => (
-      txCount + block.transactions.length
-    ), 0);
-    const missingNumBlocks = this.numberOfBlocks - numBlocksInProps;
-    return missingNumBlocks + numTxsInPropsBlocks;
+    if (this._estNumberOfTxs === undefined) {
+      const blocks = this.props.blocks;
+      const numBlocksInProps = blocks.length;
+      const numTxsInPropsBlocks = blocks.reduce(
+        (txCount, block) => txCount + block.transactions.length,
+        0
+      );
+      const missingNumBlocks = this.numberOfBlocks - numBlocksInProps;
+      this._estNumberOfTxs = missingNumBlocks + numTxsInPropsBlocks;
+    }
+    return this._estNumberOfTxs;
+  }
+
+  get numberOfPages() {
+    if (this._numberOfPages === undefined) {
+      this._numberOfPages = Math.ceil(
+        this.estNumberOfTxs / this.numTxsToDisplay
+      );
+    }
+    return this._numberOfPages;
   }
 
-  getNumberOfPages() {
-    return Math.ceil(this.estNumberOfTxs / this.numTxsToDisplay);
+  resetNums() {
+    this._numberOfBlocks = undefined;
+    this._estNumberOfTxs = undefined;
+    this._numberOfPages = undefined;
```

Comment:
Just a semantic question: Shouldn't these values be initialized (and reset) to `0` instead of `undefined` ? While it often makes sense in JS to start with and check for `undefined` variables, I think as soon as a routine is introduced that resets variables, it most likely means that these variables have a proper initial value other than `undefined`.

Alternatively, if the variables do need a value for reset, but for whatever reason can't be `0`, we should at least reset (and initialize them) with `null` instead. That would probably affect some conditional checks in the code above. 

Would be great if we could avoid re-initializing variables with `undefined` (and `null`) altogether!

Reply:
I'm using the `this._` properties for *ad hoc* memoization of values calculated by the corresponding getters. Each time `render` is called, I want to reset the getters to an un-calculated state, so I need to use `undefined` or `null` or some other unique value that can't be mistaken for a calculated value. In this case, I don't see any advantage of `null` over `undefined`, but maybe I'm missing something. Thoughts?

Reply:
Okay, understand. I'm not sure how it will affect your code, it's just that at a first glance, those variables look like they want to be `0` as initialized/reset value.

>  In this case, I don't see any advantage of null over undefined, but maybe I'm missing something. Thoughts?

Nope, you're right, there's no technical advantage. Just semantics. While `undefined` is what a variable is when it's declared by not initialized, `null` represents a value that isn't anything concrete, but still a value (which is what you want).

Reply:
JS is a rather quirky when it comes to initialization. For example, in a node REPL or browser console:

```
> x
ReferenceError: x is not defined
> typeof x
'undefined'
> let x;
undefined
> typeof x
'undefined'
```

So distinguishing not-defined variables from unitialized ones is a little tricky. When it comes to object properties, depending on whether it's important or not to consider props on the prototype chain, one can reach for `.hasOwnProperty()` to distinguish a missing property from the value `undefined`.

If I switch it to use `null` in `resetNums` then it will be important to either call `resetNums` in the constructor or duplicate (in the constructor) the lines setting the `this._` properties to `null`.

Alternatively, to avoid extra code in the constructor (it's not a big deal, I'm just exploring what's possible), since `!=`/`==` is [safe to use](https://dorey.github.io/JavaScript-Equality-Table/) when you're only concerned with `null` and `undefined`, lines such as `if (this._estNumberOfTxs === undefined) {` could be rewritten as `if (this._estNumberOfTxs == null) {` or `if (this._estNumberOfTxs == undefined) {`. With either rewrite, the expression would only be true if the value is `null` or `undefined`.  A downside is that the linter would complain about `==` and would have to be disabled for that line.

In summary, I'd like to stick with the current implementation, but if you feel using `null` helps the semantics a lot, I'll gladly make the change, and will opt to call `resetNums` in the constructor.

Reply:
You can also check for `=== null`.

Either way is fine. You can keep as is. It's just something that looks like it wants to be changed haha.

Reply:
I agree that `null`  is a little better than `undefined`, but not a huge deal.

Reply:
Okay, you've convinced me, I'll change it. :)

Reply:
> You can also check for `=== null`

Of course. I meant that by checking `== null` don't have to set the `this._` props to `null` in the constructor.

In any case, done, changes made.